core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -1425,12 +1427,23 @@ private void moveWhereInsideSubQueries(QueryModel model) throws SqlException {
                 literalCollector.resetNullCount();
                 traversalAlgo.traverse(node, literalCollector.lhs());
 
+
+                //todo: do not create new object ... make it zero GC
+                SortedSet<Integer> literalCollectorsSet = new TreeSet<>();
```

Comment:
lets replace this

Reply:
sorry, pushed my changes to diff branch, will close this PR